vault/token_store.go

```diff
@@ -530,23 +648,38 @@ func (ts *TokenStore) revokeTreeSalted(saltedId string) error {
 	return nil
 }
 
+// handleCreateRole handles the auth/token/create path for a role
+func (ts *TokenStore) handleCreateRole(
```

Comment:
This name confuses me. Can we rename it to something which indicates that this is not role creation, instead this is token creation using the role?